### ISSUE TITLE
Question about GraphSAGE and PinSAGE implementation

### ISSUE THREAD

**Issue by HiromasaYamanishi** [massquantity/LibRecommender#231]
Thank you for your great work. I am refering to your work now.
I have a question about the implementations about GraphSAGE and PinSAGE. The Question Lies in forward function in GraphSAGEModel specifically line 90-105 (https://github.com/massquantity/LibRecommender/blob/master/libreco/algorithms/torch_modules/graphsage_module.py) and PinSAGEModel line 97-113 (https://github.com/massquantity/LibRecommender/blob/master/libreco/algorithms/torch_modules/pinsage_module.py)

To my understanding, layer wise propagation of GraphSAGE and PinSAGE don't need inner loop of (for k in range(depth)) and outer loop is only necessary.
In other words, original paper propagates like h(0)→h(1)→h(2)→h(3). Although It seems like it propagates h(0)→h(1)→h(2)→h(3), h(1)→h(2)→h(3), h(2)→h(3) in case of num layer is 3.

Sorry If am I misunderstanding, I'm looking forward the reply.

**Comment by HiromasaYamanishi** [comment]
I understand the True meaning of implementation now. Sorry!
